tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Analyzer.scala

```diff
@@ -336,20 +336,94 @@ final class Analyzer(semantics: Semantics, outputMode: OutputMode,
     }
 
     def tryLookupMethod(methodName: String): Option[MethodInfo] = {
-      assert(isScalaClass,
+      assert(isScalaClass || isInterface,
           s"Cannot call lookupMethod($methodName) on non Scala class $this")
+
       @tailrec
-      def loop(ancestorInfo: ClassInfo): Option[MethodInfo] = {
+      def tryLookupInherited(ancestorInfo: ClassInfo): Option[MethodInfo] = {
         if (ancestorInfo ne null) {
           ancestorInfo.methodInfos.get(methodName) match {
             case Some(m) if !m.isAbstract => Some(m)
-            case _ => loop(ancestorInfo.superClass)
+            case _ => tryLookupInherited(ancestorInfo.superClass)
           }
         } else {
           None
         }
       }
-      loop(this)
+      val existing =
+        if (isScalaClass) tryLookupInherited(this)
+        else methodInfos.get(methodName).filter(!_.isAbstract)
+
+      if (!allowAddingSyntheticMethods) {
+        existing
+      } else if (existing.exists(m => !m.isDefaultBridge || m.owner == this)) {
+        /* If we found a non-bridge, it must be the right target.
+         * If we found a bridge directly in this class/interface, it must also
+         * be the right target.
+         */
+        existing
+      } else {
+        // Try and find the target of a possible default bridge
+        findDefaultTarget(methodName).fold {
+          existing
```

Comment:
Shouldn't we be sure that `existing` is `None` here? By contradiction, `existing = Some(m)`. If `!m.isDefaultBridge`, we should never end up here. Therefore, `m.isDefaultBridge`. But we didn't find a default target from the subclass. That shouldn't happen.

Unless that reasoning is wrong, consider `assert(existing.isEmpty)`.